weblate/trans/tests/test_models.py

```diff
@@ -334,6 +335,37 @@ def test_flags(self):
         unit.flags = "no-wrap, ignore-same"
         self.assertEqual(unit.all_flags.items(), {"no-wrap", "ignore-same"})
 
+    def test_order_by_request(self):
+        unit = Unit.objects.filter(translation__language_code="cs")[0]
+        source = unit.source_info
+        source.extra_flags = "priority:200"
+        source.save()
+
+        # test both ascending and descending order works
+        unit1 = Unit.objects.filter(translation__language_code="cs")
+        unit1 = unit1.order_by_request({"sort_by": "-priority"})
+        self.assertEqual(unit1[0].priority, 200)
+        unit1 = Unit.objects.filter(translation__language_code="cs")
+        unit1 = unit1.order_by_request({"sort_by": "priority"})
+        self.assertEqual(unit1[0].priority, 100)
+
+        # test if invalid sorting, then sorted in default order
+        unit2 = Unit.objects.filter(translation__language_code="cs")
+        unit2 = unit2.order()
+        unit3 = Unit.objects.filter(translation__language_code="cs")
+        unit3 = unit3.order_by_request({"sort_by": "invalid"})
+        self.assertEqual(unit3[0], unit2[0])
+
+        # test sorting by count
+        unit4 = Unit.objects.filter(translation__language_code="cs")[2]
+        Comment.objects.create(unit=unit4, comment="Foo")
+        unit5 = Unit.objects.filter(translation__language_code="cs")
+        unit5 = unit5.order_by_request({"sort_by": "-num_comments"})
+        self.assertEqual(unit5[0].comment_set.count(), 1)
+        unit5 = Unit.objects.filter(translation__language_code="cs")
+        unit5 = unit5.order_by_request({"sort_by": "num_comments"})
+        self.assertEqual(unit5[0].comment_set.count(), 0)
```

Comment:
Can you please test all ordering options? At least to the level that the generated queries can be executed, it doesn't have to cover validation of the ordering being correct.

Reply:
Have added a simple test to loop over all the ordering options and check if the queryset count still works. I hope that kind of ensures what you were asking. Also, I have added a similar test to check that multiple ordering options also doesn't cause any error. Let me know if this seems good.